lib/types/zpool.nix

```diff
@@ -134,13 +160,16 @@ in
       inherit config options;
       default =
         let
-          formatOutput = mode: members: ''
-            entries+=("${mode}=${
+          formatOutput = type: mode: members: ''
+            entries+=("${type} ${mode}=${
               lib.concatMapStringsSep " "
               (d: if lib.strings.hasPrefix "/" d then d else "/dev/disk/by-partlabel/disk-${d}-zfs") members
             }")
           '';
-          formatVdev = vdev: formatOutput vdev.mode vdev.members;
+          formatVdev = type: vdev: formatOutput type vdev.mode vdev.members;
+          formatVdevList = type: vdevs: lib.concatMapStrings
+            (formatVdev type)
+            (builtins.sort (a: b: a.mode < b.mode) vdevs);
```

Comment:
Is it normal for programming languages to sort the empty string first? Otherwise, a comment might be useful to communicate how much lifting the `<` does.

Reply:
I believe so, yes. In general, a word ending would be sorted before all words that start with the same letters and then continue on (e.g. `a` is before `apple`). This is also how Wikipedia defines an alphabetical sort.

That being said, I think I'll adjust this so that the empty string is first and everything else is left in the order defined by the user. This makes it more explicit, and makes the resulting order more closely match what the user defined.